From 1d89ee0b0e41c92830e59d5a7c2b59a61d4b3899 Mon Sep 17 00:00:00 2001
From: Rohit Athavale <rathaval@xilinx.com>
Date: Mon, 5 Mar 2018 16:03:48 -0800
Subject: [PATCH 0314/1852] v4l: xilinx-tpg: Add YUV 420 media bus format
 support to TPG driver

commit bfca57161dfed15bc4d4f34899ff1b08fdfab0c3 from
https://github.com/Xilinx/linux-xlnx.git

This patch updates the TPG IP driver to add support for the newly
created YUV 420 bus format.

Signed-off-by: Rohit Athavale <rathaval@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index d9ebf3cf104f..5495a5607e1d 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -259,6 +259,9 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 		u32 fmt = 0;
 
 		switch (xtpg->vip_format->code) {
+		case MEDIA_BUS_FMT_VYYUYY8_1X24:
+			fmt = XTPG_HLS_COLOR_FORMAT_YUV_420;
+			break;
 		case MEDIA_BUS_FMT_UYVY8_1X16:
 			fmt = XTPG_HLS_COLOR_FORMAT_YUV_422;
 			break;
-- 
2.31.1

